pkg/storaged/resize.jsx

```diff
@@ -0,0 +1,450 @@
+            }
+        }
+        return false;
+    }
+
+    console.log("F", find_it(parts));
+    return find_it(parts) || 0;
+}
+
+export function grow_dialog(client, lvol_or_part, info, to_fit) {
+    let title, block, name, orig_size, max_size, allow_infinite, round_size;
+
+    if (lvol_or_part.iface == "org.freedesktop.UDisks2.LogicalVolume") {
+        const vgroup = client.vgroups[lvol_or_part.VolumeGroup];
+        const pool = client.lvols[lvol_or_part.ThinPool];
+
+        title = _("Grow logical volume");
+        block = client.lvols_block[lvol_or_part.path];
+        name = lvol_or_part.Name;
+        orig_size = lvol_or_part.Size;
+        max_size = pool ? pool.Size * 3 : lvol_or_part.Size + vgroup.FreeSize;
+        allow_infinite = !!pool;
+        round_size = vgroup.ExtentSize;
+    } else {
+        title = _("Grow partition");
+        block = client.blocks[lvol_or_part.path];
+        name = block_name(block);
+        orig_size = lvol_or_part.Size;
+        max_size = lvol_or_part.Size + free_space_after_part(client, lvol_or_part);
+        allow_infinite = false;
+        round_size = 1024 * 1024;
+    }
+
+    const usage = get_active_usage(client, block && info.grow_needs_unmount ? block.path : null, _("grow"));
+
+    if (usage.Blocking) {
+        dialog_open({
+            Title: cockpit.format(_("$0 is in use"), name),
+            Body: BlockingMessage(usage)
+        });
+        return;
+    }
+
+    let grow_size;
+    let size_fields = [];
+    if (!to_fit) {
+        size_fields = [
+            SizeSlider("size", _("Size"),
+                       {
+                           value: orig_size,
+                           min: orig_size,
+                           max: max_size,
+                           allow_infinite,
+                           round: round_size,
+                       })
+        ];
+    } else {
+        grow_size = block.Size;
+    }
+
+    let recovered_passphrase;
+    let passphrase_fields = [];
+    if (block && block.IdType == "crypto_LUKS" && block.IdVersion == 2)
+        passphrase_fields = existing_passphrase_fields(_("Resizing an encrypted filesystem requires unlocking the disk. Please provide a current disk passphrase."));
+
+    if (!usage.Teardown && size_fields.length + passphrase_fields.length === 0) {
+        return lvol_or_part_and_fsys_resize(client, lvol_or_part, grow_size, info.grow_needs_unmount, null);
+    }
+
+    const dlg = dialog_open({
+        Title: title,
+        Teardown: TeardownMessage(usage),
+        Fields: size_fields.concat(passphrase_fields),
+        Action: {
+            Title: _("Grow"),
+            action: function (vals) {
+                return teardown_active_usage(client, usage)
+                        .then(function () {
+                            return (lvol_or_part_and_fsys_resize(client, lvol_or_part,
+                                                                 to_fit ? grow_size : vals.size,
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19385-20230925-111150-44391ef2-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)